src/axom/sidre/tests/sidre_mfem_datacollection.cpp

```diff
@@ -261,6 +259,197 @@ TEST(sidre_datacollection, dc_alloc_nonowning_parmesh)
+
+  MFEMSidreDataCollection sdc_reader(testName());
+
+  // Needs to be set "manually" in order for everything to be loaded in properly
+  sdc_reader.SetComm(MPI_COMM_WORLD);
+  sdc_reader.Load();
+
+  // Make sure the mesh was actually reconstructed
+  EXPECT_EQ(sdc_reader.GetMesh()->GetNV(), n_verts);
+  EXPECT_EQ(sdc_reader.GetMesh()->GetNE(), n_ele);
+  EXPECT_EQ(sdc_reader.GetMesh()->GetNBE(), n_bdr_ele);
+
+  // Make sure the ParMesh was reconstructed - even on one rank, this
+  // will still be a ParMesh
+  auto reader_pmesh = dynamic_cast<mfem::ParMesh*>(sdc_reader.GetMesh());
+  ASSERT_NE(reader_pmesh, nullptr);
+  EXPECT_EQ(reader_pmesh->GetNGroups(), n_groups);
+  EXPECT_EQ(reader_pmesh->GetNFaceNeighbors(), n_face_neighbors);
+  EXPECT_EQ(reader_pmesh->GetNSharedFaces(), n_shared_faces);
+
+  auto reader_group_data = getGroupData(*reader_pmesh);
+  EXPECT_EQ(writer_group_data, reader_group_data);
+  EXPECT_TRUE(sdc_reader.verifyMeshBlueprint());
+}
+
+/**
+ * @brief Wrapper for testing mesh reconstruction across all partitioning methods
+ * @param [in] base_mesh The serial mesh object to distribute, save, and then reload
+ */
+static void testParallelMeshReloadAllPartitionings(mfem::Mesh& base_mesh)
+{
+  for(int part_method = 0; part_method <= 5; part_method++)
+  {
+    testParallelMeshReload(base_mesh, part_method);
+  }
+}
+
+TEST(sidre_datacollection, dc_par_reload_gf)
+{
+  const std::string field_name = "test_field";
+  // 3D tet mesh
+  mfem::Mesh mesh(2, 2, 2, mfem::Element::TETRAHEDRON);
+  mfem::ParMesh parmesh(MPI_COMM_WORLD, mesh);
+
+  mfem::H1_FECollection fec(1, mesh.Dimension());
+  mfem::ParFiniteElementSpace parfes(&parmesh, &fec);
+
+  // The mesh must be owned by Sidre to properly manage data in case of
+  // a simulated restart (save -> load)
+  bool owns_mesh = true;
+  MFEMSidreDataCollection sdc_writer(testName(), &parmesh, owns_mesh);
+
+  // The mesh and field(s) must be owned by Sidre to properly manage data in case of
+  // a simulated restart (save -> load)
+  mfem::ParGridFunction gf_write(&parfes, static_cast<double*>(nullptr));
+
+  // Register to allocate storage internally, then write to it
+  sdc_writer.RegisterField(field_name, &gf_write);
+
+  mfem::ConstantCoefficient three_and_a_half(3.5);
+  gf_write.ProjectCoefficient(three_and_a_half);
+
+  sdc_writer.SetCycle(0);
+  sdc_writer.Save();
+
+  MFEMSidreDataCollection sdc_reader(testName());
+
+  // Needs to be set "manually" in order for everything to be loaded in properly
+  sdc_reader.SetComm(MPI_COMM_WORLD);
+  sdc_reader.Load();
+
+  auto gf_read = sdc_reader.GetField(field_name);
+  EXPECT_TRUE(dynamic_cast<mfem::ParGridFunction*>(gf_read));
+
+  // Make sure the gridfunction was actually read in
+  EXPECT_LT(gf_read->ComputeL2Error(three_and_a_half), EPSILON);
+
+  EXPECT_TRUE(sdc_reader.verifyMeshBlueprint());
+}
+
+// MFEM's partitioning logic looks like it's buggy for 1D meshes
+// Probably not a problem since 1D meshes are rare
```

Comment:
> // MFEM's partitioning logic looks like it's buggy for 1D meshes
// Probably not a problem since 1D meshes are rare 

Please open an mfem ticket for this.

Reply:
It looks like my comment is actually incorrect - I was able to add a 1-dimensional mesh test with some minor changes